tests/end_to_end/test_account.py

```diff
@@ -34,54 +56,57 @@ async def test_should_upgrade_outdated_account_on_transfer(
-            erc20_token,
+            counter,
             new_account,
-            other,
             class_hashes,
         ):
-            # Given an initial class
             prev_class = await starknet.get_class_hash_at(
                 new_account.starknet_contract.address
             )
             target_class = class_hashes["account_contract_fixture"]
             assert prev_class != target_class
             assert prev_class == class_hashes["account_contract"]
 
-            # When setting the account class to version `account_contract_fixture`
             await invoke(
                 "kakarot",
                 "set_account_contract_class_hash",
                 target_class,
             )
 
-            # Then the account should process the next transaction and be upgraded to the new class
-            receipt = (
-                await erc20_token.transfer(
-                    other.address, TEST_AMOUNT, caller_eoa=new_account.starknet_contract
-                )
-            )["receipt"]
-            events = erc20_token.events.parse_starknet_events(receipt.events)
-            assert events["Transfer"] == [
-                {
-                    "from": new_account.address,
-                    "to": other.address,
-                    "value": TEST_AMOUNT,
-                }
-            ]
-            assert (
-                await erc20_token.balanceOf(new_account.address)
-                == TOTAL_SUPPLY - TEST_AMOUNT
-            )
-            assert await erc20_token.balanceOf(other.address) == TEST_AMOUNT
+            await assert_counter_transaction_success(counter, new_account)
 
             new_class = await starknet.get_class_hash_at(
                 new_account.starknet_contract.address
             )
             assert new_class == target_class
 
-            # Reset the account class to the previous version
+        async def test_should_update_cairo1_helpers_class(
+            self,
+            starknet: FullNodeClient,
+            invoke,
+            counter,
+            new_account,
+            class_hashes,
+        ):
+            prev_cairo1_helpers_class = await starknet.get_storage_at(
+                new_account.starknet_contract.address,
+                get_storage_var_address("Account_cairo1_helpers_class_hash"),
+            )
+            target_class = class_hashes["Cairo1HelpersFixture"]
+            assert prev_cairo1_helpers_class != target_class
+            assert prev_cairo1_helpers_class == class_hashes["Cairo1Helpers"]
+
             await invoke(
                 "kakarot",
                 "set_account_contract_class_hash",
-                prev_class,
+                class_hashes["account_contract_fixture"],
+            )
+            await invoke("kakarot", "set_cairo1_helpers_class_hash", target_class)
+
+            await assert_counter_transaction_success(counter, new_account)
+
+            new_cairo1_helpers_class = await starknet.get_storage_at(
```

Comment:
you can do `assert await xxx == target_class` to avoid naming a variable and make explicit that the assert is about a raw "on-chain" info